### ISSUE TITLE
Question --- Possible to grab partial md5 for save names?

### ISSUE THREAD

**Issue by Aelphais** [Bionus/imgbrd-grabber#1355]
As it turns out, saving billions of files in one folder causes some issues. Most notably the folder load time if I try to browse to it via windows explorer. To alleviate this issue, I've split off the images into subdirectories named after the first two letters of the hash. So for example: 4b0c07336d92532b80d113b84df064f7.png would be in folder 4b.

I have a php script set up to generate image thumbnails that I run after downloading any batches. At the same time as it makes the thumbnail, it moves the master image to the subdirectory. This works fine and, thanks to the "If a file already exists" feature, I don't end up downloading duplicates. However, moving the images does cause a nearly insignificant slowdown compared to before when everything downloaded to one folder.

So I was wondering if I could get around this by downloading straight to the subdirectory folders. Perhaps some filename formatting token like %md5-x% where x is the number of characters from 0 that I want to substring? I'm not sure if something like that already exists or if it is even feasible. I looked through the available tokens list and didn't see anything, so maybe this is more of a feature request?

Anyway, thanks for all your hard work regardless! Grabber is incredibly handy!

tl;dr: I want to substring the %md5% token during saving to split images off into different folders during download.

**Comment by Bionus** [comment]
Yes, its possible to do so!

You can use the `maxlength` token option:
https://github.com/Bionus/imgbrd-grabber/wiki/Filename#maxlength-int

So in your example, the filename would be:
```
%md5:maxlength=2%/%md5%.%ext%
```

**Comment by Aelphais** [comment]
Ah, I see I just didn't scroll far enough down! Thanks again!
